pgcli/pgexecute.py

```diff
@@ -17,6 +17,27 @@
 )
 
 
+# we added this funcion to strip beginning comments
+# because sqlparse didn't handle tem well.  It won't be needed if sqlparse
+# does parsing of this situation better
+
+
+def remove_beginning_comments(command):
+    # Regular expression pattern to match comments
+    pattern = r"^(/\*.*?\*/|--.*?)(?:\n|$)"
```

Comment:
I think that `sqlparse.format(command, strip_comments=True)` may do exactly what you want. I didn't try it myself, though, so take it with a grain of salt, but I suggest trying.

Reply:
Thank you for taking time to analyze this.  If I remember correctly, we decided to not use the strip_comments flags in sqlparse because it truly throws away the comments and some user wanted to comments to show up in postgres error/output logs.  So this "dance" in the code is to save the beginning comments, remove them from the sql query, parse the sql query, then re-add the comments to the beginning of the query so that they show up in error/output logs.

Does that make sense or would you like to tackle this a different way.

Reply:
@j-bennet Here is the original discussion about not stripping comments [https://github.com/dbcli/pgcli/issues/1286](https://github.com/dbcli/pgcli/issues/1286)

Reply:
I see, it makes a lot more sense now. I forgot that we had a reason to preserve comments.

Ok, so if psycopg is able to handle statements with comments with them, why do we get this error?